jena-text/src/main/java/org/apache/jena/query/text/TextIndexLucene.java

```diff
@@ -722,6 +723,10 @@ private String composeQField(String qs, String textField, String lang, boolean u
         List<String> searchForTags = Util.getSearchForTags(lang);
         boolean usingSearchFor = Util.usingSearchFor(lang);
 
+        // The set will reduce lucene text field expressions
+        // to unique expressions
+        Set<String> uniquePropListQueryStrings = new HashSet<>();
```

Comment:
Is there any scenario in which the ordering of the generated clauses is going to matter?

i.e. should this be a `LinkedHashSet` instead?

+1 on eliminating duplicate clauses

Reply:
Changed to `LinkedHashSet`